screen2.0/src/app/_scatterPlot/scatterPlot.tsx

```diff
@@ -466,12 +491,14 @@ function Umap({ width: parentWidth, height: parentHeight, pointData: umapData, l
                             {tooltipOpen && tooltipData && isHoveredPointWithinBounds &&(
                                 <Tooltip left={xScaleTransformed(tooltipData.x) + 50} top={yScaleTransformed(tooltipData.y) + 50}>
                                     <div>
-                                        <strong>Name: </strong> 
-                                        {tooltipData.name.replace(/_/g, " ").slice(0, 45)}
-                                        {tooltipData.name.length > 45 ? "..." : ""}
-                                    </div>
-                                    <div>
-                                        <strong>Accession:</strong> {tooltipData.accession}
+                                        {Object.entries(tooltipData.metaData).map(([key, value]) => (
+                                            <div key={key}>
+                                                <strong>{key.charAt(0).toUpperCase() + key.slice(1)}: </strong> 
+                                                {typeof value === "string" && value.length > 45 
```

Comment:
since the first typeof check will be false for numbers, won't numbers never show up?